File: flutter/measure_flutter/android/src/main/kotlin/sh/measure/flutter/MethodConstants.kt
package sh.measure.flutter

object MethodConstants {
    // Function names
    const val FUNCTION_TRACK_CUSTOM_EVENT = "trackCustomEvent"
    const val FUNCTION_TRACK_EXCEPTION = "trackException"
    const val FUNCTION_NATIVE_CRASH = "triggerNativeCrash"

    // Argument keys
    const val ARG_NAME = "name"
    const val ARG_TIMESTAMP = "timestamp"
    const val ARG_ATTRIBUTES = "attributes"
    const val ARG_SERIALIZED_EXCEPTION = "serialized_exception"


    // Exception object
    const val EXCEPTION_EXCEPTIONS = "exceptions"
    const val EXCEPTION_HANDLED = "handled"
    const val EXCEPTION_TYPE = "type"
    const val EXCEPTION_MESSAGE = "message"
    const val EXCEPTION_FRAMES = "frames"
    const val EXCEPTION_FRAME_CLASS_NAME = "class_name"
    const val EXCEPTION_FRAME_METHOD_NAME = "method_name"
    const val EXCEPTION_FRAME_FILE_NAME = "file_name"
    const val EXCEPTION_FRAME_LINE_NUM = "line_num"
    const val EXCEPTION_FRAME_MODULE_NAME = "module_name"
    const val EXCEPTION_FRAME_COL_NUM = "col_num"
    const val EXCEPTION_FRAME_INDEX = "index"
    const val EXCEPTION_FRAME_BINARY_ADDR = "binary_addr"
    const val EXCEPTION_FRAME_INSTRUCTION_ADDR = "instruction_addr"

    // Error codes
    const val ERROR_INVALID_ARGUMENT = "invalid_argument"
    const val ERROR_ARGUMENT_MISSING = "argument_missing"
    const val ERROR_INVALID_ATTRIBUTE = "invalid_attribute"
    const val ERROR_UNKNOWN = "unknown_error"
}

File: flutter/measure_flutter/ios/Classes/MethodConstants.swift
//
//  MethodConstants.swift
//  measure_flutter
//
//  Created by Abhay Sood on 12/02/25.
//

import Foundation

enum MethodConstants {
    // Function names
    static let functionTrackCustomEvent = "trackCustomEvent"
    static let functionTrackException = "trackException"
    static let functionNativeCrash = "triggerNativeCrash"

    // Argument keys
    static let argName = "name"
    static let argTimestamp = "timestamp"
    static let argAttributes = "attributes"
    static let argSerializedException = "serialized_exception"

    // Exception object
    static let exceptionExceptions = "exceptions"
    static let exceptionHandled = "handled"
    static let exceptionType = "type"
    static let exceptionMessage = "message"
    static let exceptionFrames = "frames"
    static let exceptionFrameClassName = "class_name"
    static let exceptionFrameMethodName = "method_name"
    static let exceptionFrameFileName = "file_name"
    static let exceptionFrameLineNum = "line_num"
    static let exceptionFrameModuleName = "module_name"
    static let exceptionFrameColNum = "col_num"
    static let exceptionFrameIndex = "index"
    static let exceptionFrameBinaryAddr = "binary_addr"
    static let exceptionFrameInstructionAddr = "instruction_addr"

    // Error codes
    static let errorInvalidArgument = "invalid_argument"
    static let errorArgumentMissing = "argument_missing"
    static let errorInvalidAttribute = "invalid_attribute"
    static let errorUnknown = "unknown_error"
}
